rel/parse_filterql.go

```diff
@@ -11,85 +11,106 @@ import (
 	"github.com/araddon/qlbridge/lex"
 )
 
-// ParseFilterQL Parses a FilterQL statement
-func ParseFilterQL(filter string) (*FilterStatement, error) {
-	l := lex.NewFilterQLLexer(filter)
-	m := filterQLParser{l: l, filterTokenPager: newFilterTokenPager(l), buildVm: false}
-	return m.parseFilterStart()
+type FilterQLParser struct {
+	statement string //can be a FilterStatement, FilterStatements, filterSelect, filterSelects, etc.  Which one is determined by which Parser func you call.
+
+	buildVm bool
+	fs      *FilterStatement
+	l       *lex.Lexer
+	comment string
+	*filterTokenPager
+	firstToken lex.Token
+
+	funcs expr.FuncResolver
 }
 
```

Comment:
not sure if you like the builder pattern `NewFilterParser().Statement(filter).ParseFilter()`.  we can change it if you don't.